src/compiler/sys.ts

```diff
@@ -16,6 +16,16 @@ namespace ts {
         }
     }
 
+    /**
+     * djb2 hashing algorithm
+     * http://www.cse.yorku.ca/~oz/hash.html
+     */
+    /* @internal */
+    export function generateDjb2Hash(data: string): string {

```

Comment:
no need for this to be exported i would say. just make createHash always implemented in sys.